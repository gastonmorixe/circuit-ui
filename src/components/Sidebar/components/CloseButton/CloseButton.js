import styled, { css } from 'react-emotion';
import PropTypes from 'prop-types';

const baseStyles = ({ theme, visible }) => css`
  outline: none;
  height: 40px;
  width: 40px;
  border-radius: 50%;
  background-color: ${theme.colors.n100};
  position: absolute;
  bottom: 16px;
  right: 16px;
  transition: opacity 150ms ease-in-out, visibility 150ms ease-in-out;
  visibility: ${visible ? 'visible' : 'hidden'};
  opacity: ${visible ? 1 : 0};
  z-index: 1;
  ${theme.mq.mega`
    visibility: hidden;  
  `};
`;

const CloseButton = styled('button')(baseStyles);

CloseButton.propTypes = {
  /**
   * Tells if the CloseButton is visible
   */
  visible: PropTypes.bool
};

export default CloseButton;
