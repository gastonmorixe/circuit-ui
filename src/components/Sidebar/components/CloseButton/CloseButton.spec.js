import React from 'react';

import CloseButton from './CloseButton';

describe('CloseButton', () => {
  describe('styles', () => {
    it('should render and match snapshot when not visible', () => {
      const actual = create(<CloseButton visible={false} />);
      expect(actual).toMatchSnapshot();
    });

    it('should render and match snapshot when visible', () => {
      const actual = create(<CloseButton visible={true} />);
      expect(actual).toMatchSnapshot();
    });
  });

  describe('accessibility', () => {
    it('should meet accessibility guidelines', async () => {
      const wrapper = renderToHtml(<CloseButton />);
      const actual = await axe(wrapper);
      expect(actual).toHaveNoViolations();
    });
  });
});
