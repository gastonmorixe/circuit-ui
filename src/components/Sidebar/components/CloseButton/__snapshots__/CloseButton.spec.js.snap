// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton styles should render and match snapshot when not visible 1`] = `
.circuit-0 {
  outline: none;
  height: 40px;
  width: 40px;
  border-radius: 50%;
  background-color: #FAFBFC;
  position: absolute;
  bottom: 16px;
  right: 16px;
  -webkit-transition: opacity 150ms ease-in-out,visibility 150ms ease-in-out;
  transition: opacity 150ms ease-in-out,visibility 150ms ease-in-out;
  visibility: hidden;
  opacity: 0;
  z-index: 1;
}

@media (min-width:768px) {
  .circuit-0 {
    visibility: hidden;
  }
}

<button
  className="circuit-0 circuit-1"
/>
`;

exports[`CloseButton styles should render and match snapshot when visible 1`] = `
.circuit-0 {
  outline: none;
  height: 40px;
  width: 40px;
  border-radius: 50%;
  background-color: #FAFBFC;
  position: absolute;
  bottom: 16px;
  right: 16px;
  -webkit-transition: opacity 150ms ease-in-out,visibility 150ms ease-in-out;
  transition: opacity 150ms ease-in-out,visibility 150ms ease-in-out;
  visibility: visible;
  opacity: 1;
  z-index: 1;
}

@media (min-width:768px) {
  .circuit-0 {
    visibility: hidden;
  }
}

<button
  className="circuit-0 circuit-1"
/>
`;
